### ISSUE TITLE
QUnit crashes if promise is rejected without arguments

### ISSUE THREAD

**Issue by tbiesemann** [jquery/qunit#1039]
In case of a failing test when a promise gets rejected, QUnit crashes internally with a  null reference.

This can be reproduced using this snippet:

```js
	QUnit.test("QUnit shall handle failing tests properly", function(assert) {
		var resolve = function(){};
		var reject = function(){};
		var promise = new Promise(function(_resolve, _reject) {
			reject = _reject;
		});
		reject(undefined);
		return promise;
	});
```

The crash happens here:
```js
	resolvePromise: function( promise, phase ) {
		var then, resume, message,
			test = this;
		if ( promise != null ) {
			then = promise.then;
			if ( QUnit.objectType( then ) === "function" ) {
				resume = internalStop( test );
				then.call(
					promise,
					function() { resume(); },
					function( error ) {
						message = "Promise rejected " +
							( !phase ? "during" : phase.replace( /Each$/, "" ) ) +
							" " + test.testName + ": " + ( error.message || error );
						test.pushFailure( message, extractStacktrace( error, 0 ) );
						// Else next test will carry the responsibility
						saveGlobal();
						// Unblock
						resume();
					}
				);
			}
		}
	},
```

The error object is undefined, hence error.message will fail.

Best, Thomas

**Comment by platinumazure** [comment]
I'll investigate this weekend, unless someone beats me to it. Thanks for the report!

**Comment by platinumazure** [comment]
@tbiesemann I've edited the original post to preserve your whitespace and code formatting a little better-- hope that's okay. In future, you can just use three backticks and a language identifier ("js" for JavaScript) and then your spacing is preserved.

**Comment by trentmwillis** [comment]
Fixed via #1041.
